src/Microsoft.Azure.SignalR.Protocols/CheckWithAckMessage.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.SignalR.Protocol
+{
+    /// <summary>
+    /// Base class of check-with-ack messages between Azure SignalR Service and SDK.
+    /// </summary>
+    public abstract class CheckWithAckMessage : ExtensibleServiceMessage, IAckableMessage, IMessageWithTracingId
```

Comment:
Given that all the `Check*Message` must have ack, no need to add `WithAck` to the name?

Reply:
Since we have non ackable and ackable message, messages implement `IAckableMessage` has the suffix `WithAckMessage`, so I follow the naming convention

Reply:
I think `WithAck` is necessary when we have the same kind messages with ack and without ack, such as `JoinGroupWithAckMessage` and `JoinGroupMessage`. For `Check*Message`, we only have messages with ack, so there is no need to append `WithAck` to the class name, which seems redundant.

Reply:
Let's use `CheckUserExists/CheckGroupExists/CheckConnectionExists` to be consistent with REST API definitions

Reply:
If we didn't already have `*GroupWithAckMessage` I'd propose `AckableCheckMessage` or `AckableQueryMessage`. 

How about have the `ack` in for the base class and not for the derived ones?

Reply:
I totally agree we can have a better naming if the `ackable` message is added at 1st time, but we already have messages with suffix `WithAckMessage`, e.g. `JoinGroupWithAckMessage`, I think following the naming convention is better.

Reply:
actually, out rest api doc doesn't have the consistency for now, e.g. `Check if there are any client connections connected for the given user`, maybe we can also update it in the future